src/sentry/web/api.py

```diff
@@ -329,12 +329,12 @@ def process(self, request, project, auth, helper, data, **kwargs):
 
         content_encoding = request.META.get('HTTP_CONTENT_ENCODING', '')
 
-        if isinstance(data, six.string_types):
+        if isinstance(data, (six.binary_type, six.text_type)):
             if content_encoding == 'gzip':
                 data = helper.decompress_gzip(data)
             elif content_encoding == 'deflate':
                 data = helper.decompress_deflate(data)
-            elif not data.startswith('{'):
+            elif data[0] != b'{':
```

Comment:
i guess this technically would break, we could do force_bytes on data[0] here

I think realistically this is *always* bytes, but I need to confirm.